libs/common/src/change-stream/test-helpers/generate-mock-audit-database-record.ts

```diff
@@ -53,3 +54,17 @@ export const generateParsedMockPortalUserAuditDatabaseRecord = (mockUserId: stri
   // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
   lastUpdatedAt: expect.any(String),
 });
+
+export const generateParsedMockAuditDatabaseRecord = (auditDetails: AuditDetails) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+  switch (auditDetails.userType) {
+    case 'tfm':
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      return generateParsedMockTfmUserAuditDatabaseRecord(auditDetails.id);
+    case 'portal':
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      return generateParsedMockPortalUserAuditDatabaseRecord(auditDetails.id);
+    default:
+      throw new Error('Invalid auditDetails userType');
+  }
+};
```

Comment:
Think we can remove the linting ignores - I've done this to the rest of the file in another PR
```suggestion
export const generateParsedMockAuditDatabaseRecord = (auditDetails: AuditDetails) => {
  switch (auditDetails.userType) {
    case 'tfm':
      return generateParsedMockTfmUserAuditDatabaseRecord(auditDetails.id) as AuditDatabaseRecord;
    case 'portal':
      return generateParsedMockPortalUserAuditDatabaseRecord(auditDetails.id) as AuditDatabaseRecord;
    default:
      throw new Error('Invalid auditDetails userType');
  }
};
```

Reply:
Depending on how we feel about saying `as AuditDatabaseRecord` when it isnt one - can alternatively use `as Object`

Reply:
Actually might not need these in this function, and just put them in the others